assertion/function/assertiontree/backprop_util.go

```diff
@@ -719,7 +719,7 @@ func exprAsDeepProducer(rootNode *RootAssertionNode, expr ast.Expr) annotation.P
 	if len(parsedExpr) > 1 {
 		panic("multiply returning function passed where a deep producer is expected - tuple types are not deep")
 	}
-	if len(parsedExpr) == 0 || !parsedExpr[0].IsDeep() {
+	if len(parsedExpr) == 0 || !parsedExpr[0].IsDeep() || parsedExpr[0].GetDeep() == nil {
```

Comment:
(fix for NilAway reported error that was reported due the support added in this PR)

Reply:
This is a wontfix false positive since `.IsDeep() => .GetDeep() != nil`, right? 

(honestly, I feel we could probably just refactor the struct to get rid of `IsDeep()`, but that should definitely happen in separate PRs

Reply:
Correct, a "wontfix" false positive. I remember I had a implemented a refactor a while ago internally on lines similar to what you are suggestions for solving this problem, I'll dig it up and create a new PR.